ToDoList/View/ListTaskViewController.swift

```diff
@@ -37,75 +37,173 @@ class ListTaskViewController: SFListPage<ListTaskViewModel> {
-        let newListTaskCellViewModel = ListTaskTableViewCellViewModel(model: newTask)
+    func add(with taskName: String, and time: Date) {
+        let newTask = TaskModel(title: taskName, time: time)
+        model?.listTask.append(newTask)
+        rxUpdateListTask.onNext(model?.listTask.count ?? 0)
+        let newListTaskCellViewModel = ListTaskTableViewCellViewModel(model: newTask, delegate: self)
         itemsSource.append(newListTaskCellViewModel)
+        let indexPath = IndexPath(row: (model?.listTask.count ?? 0) - 1, section: 0)
+        countDownTask(time, with: taskName, at: indexPath)
     }
     
     func delete(at indexPath: IndexPath) {
+        model?.listTask.remove(at: indexPath.row)
+        rxUpdateListTask.onNext(model?.listTask.count ?? 0)
         itemsSource.remove(at: indexPath)
     }
     
-    func updateListTask() {
-        countTask.accept(itemsSource.countElements())
+    func edit(at indexPath: IndexPath, with taskName: String, and time: Date) {
+        if let cellViewModel = itemsSource.element(atIndexPath: indexPath) as? ListTaskTableViewCellViewModel {
+            cellViewModel.update(text: taskName, time: time)
+        }
+        countDownTask(time, with: taskName, at: indexPath)
+    }
+    
+    func getAddEditViewModel(atIndex indexPath: IndexPath? = nil) -> AddEditTaskViewModel {
+        let addEditModel = getAddEditModel(for: indexPath)
+        return AddEditTaskViewModel(model: addEditModel, delegate: self)
+    }
+    
+    func getAddEditModel(for indexPath: IndexPath?) -> AddEditTaskModel? {
+        guard let indexPath = indexPath, let cellViewModel = itemsSource.element(atIndexPath: indexPath) as? ListTaskTableViewCellViewModel,
+              let title = cellViewModel.model?.title, let time = cellViewModel.model?.time else {
+            return nil
+        }
+        return AddEditTaskModel(indexPath: indexPath, title: title, time: time)
     }
     
-    func getAddVC() -> UIViewController {
-        let page = UIStoryboard(name: "Main", bundle: nil).instantiateViewController(withIdentifier: "AddListTaskViewController") as! AddViewController
-        let viewModel = AddViewModel()
-        page.viewModel = viewModel
-        page.viewModel?.listTaskNameSubject.subscribe(onNext: { text in
-            self.add(with: text)
-        }).disposed(by: page.disposeBag ?? DisposeBag())
-        return page
+    func countDownTask(_ time: Date, with title: String, at indexPath: IndexPath) {
+        let countDownPushNoti = Observable<Int>.timer(RxTimeInterval.seconds(time.getCountDownTimePushNoti()), scheduler: MainScheduler.instance)
+        let countDown = Observable<Int>.timer(RxTimeInterval.seconds(time.getCountDownTime()), scheduler: MainScheduler.instance)
+        guard let cellViewModel = self.itemsSource.element(atIndexPath: indexPath) as? ListTaskTableViewCellViewModel else { return }
+        DispatchQueue.main.asyncAfter(deadline: .now()) {
```

Comment:
Phần này hơi bị nhiều logic, tách function xử lý ra em. đâu cần mỗi cell một countdown đâu? hoàn toàn có thể dùng một countdown cách một khoảng thời gian lại kiểm tra một lượt các task rồi trả về trạng thái tương ứng từng task. -> để kết quả đó cho phần chịu trách nhiệm update UI theo trạng thái